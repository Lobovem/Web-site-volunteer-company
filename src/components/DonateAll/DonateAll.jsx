import { TitleMain } from '../TitleMain/TitleMain';
import privat from '../../img/icon-privat-bank.svg';
import swift from '../../img/icon-swift.svg';
import paypal from '../../img/icon-pay-pal.svg';
import mono from '../../img/icon-mono.png';
import BTC from '../../img/icon-bitcoin.svg';
import ETH from '../../img/icon-ethereum.svg';
import USDT from '../../img/icon-tether.svg';
import s from './DonateAll.module.scss';

export const DonateAll = () => {
  return (
    <div className={s.donateAll}>
      <div className={s.donateAll__wrap}>
        <TitleMain title="Маленькі кроки ведуть нас до великої ПЕРЕМОГИ!"></TitleMain>
        <h2 className={s.donateAll__title}>
          Ми не збираємо кошти і не просимо донати в будь-яких соцмережах. Все робимо своїми силами. Але, якщо хочете допомогти, ми не
          можемо відмовити вам в такому праві.
        </h2>

        <div className={s.donateAll__wrapPay}>
          <div className={s.donateAll__wrapIcon}>
            <img className={s.donateAll__img} src={privat} alt="icon-privat-bank.svg" />
            <p className={s.donateAll__text}>Рахунок (Гривня)</p>
            <p className={s.donateAll__desc}>БО МБФ УКРАЇНСЬКИЙ ГУМАНІТАРНИЙ БАТАЛЬЙОН</p>

            <p className={s.donateAll__text}>ІПН/ЄДРПОУ</p>
            <p className={s.donateAll__desc}>44746691</p>

            <p className={s.donateAll__text}>IBAN</p>
            <p className={s.donateAll__desc}>UA533052990000026004031220892</p>
          </div>

          <div className={s.donateAll__wrapIcon}>
            <img className={s.donateAll__img} src={mono} alt="icon-mono.png" />
            <p className={s.donateAll__text}>Карта (Гривна)</p>
            <p className={s.donateAll__desc}>3758640000000000</p>
          </div>

          <div className={s.donateAll__wrapIcon}>
            <img className={s.donateAll__img} src={swift} alt="icon-swift.svg" />
            <p className={s.donateAll__text}>SWIFT (Прекази у гривнях)</p>
            <p className={s.donateAll__desc}>IBAN UA00 000000 00000 00000000000000</p>
          </div>

          <div className={s.donateAll__wrapIcon}>
            <img className={s.donateAll__img} src={paypal} alt="icon-pay-pal.svg" />
            <p className={s.donateAll__text}>Перейти до PayPal</p>
            <p className={s.donateAll__desc}>namecompany@gmail.com</p>
          </div>

          <div className={s.donateAll__wrapIconAlt}>
            <img className={s.donateAll__img} src={BTC} alt="icon-bitcoin.svg" />
            <div className={s.donateAll__wrapText}>
              <p className={s.donateAll__text}>BTC</p>
              <p className={s.donateAll__desc}>0x0000000000000000000000000000000000000000</p>
            </div>

            <img className={s.donateAll__img} src={ETH} alt="icon-ethereum.svg" />
            <div className={s.donateAll__wrapText}>
              <p className={s.donateAll__text}>ETH</p>
              <p className={s.donateAll__desc}>0x00000000000000000000000000000000</p>
            </div>

            <img className={s.donateAll__img} src={USDT} alt="icon-tether.svg" />
            <div className={s.donateAll__wrapText}>
              <p className={s.donateAll__text}>USDT</p>
              <p className={s.donateAll__desc}>TG00000000000000000000000000000000</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
