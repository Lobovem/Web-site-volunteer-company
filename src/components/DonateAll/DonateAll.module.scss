@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';
@import './src/styles/MixinsKit.module.scss';

.donateAll {

  &__wrap {
    text-align: center;
  }

  &__title {
    font-weight: $font-weight-500;
    font-size: $font-size-16;
    line-height: 16px;
    color: $color-black;
    margin-bottom: 30px;

    @include lg-min {
      font-size: $font-size-24;
      line-height: 24px;
      margin-bottom: 60px;

    }
  }

  &__img {
    margin-bottom: 40px;
  }

  &__wrapPay {
    display: grid;
    grid-template-columns: repeat(1, 1fr);
    gap: 20px;

    @media (min-width:750px) {
      grid-template-columns: repeat(2, 1fr);
    }
  }

  &__wrapIcon {
    background: #FFFFFF;
    box-shadow: 0px 4px 15px rgba(2, 108, 167, 0.08);
    border-radius: 14px;
    padding: 50px;
  }

  &__wrapIconAlt {
    background: #FFFFFF;
    box-shadow: 0px 4px 15px rgba(2, 108, 167, 0.08);
    border-radius: 14px;
    padding: 50px;

    @media(min-width:750px) {
      &:last-child {
        grid-row-start: 2;
        grid-row-end: 4;
      }
    }
  }

  &__wrapText {
    margin-top: -20px;
    margin-bottom: 60px;
  }

  &__text {
    font-weight: $font-weight-500;
    font-size: $font-size-16;
    line-height: 16px;
    color: $color-black;
    margin-bottom: 5px;
  }

  &__desc {
    font-weight: $font-weight-400;
    font-size: $font-size-14;
    line-height: 16px;
    color: $color-black;
    word-break: break-word;
    margin-bottom: 15px;

    @include lg-min {
      font-size: $font-size-16;
    }
  }
}