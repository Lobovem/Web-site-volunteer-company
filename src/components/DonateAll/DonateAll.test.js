import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { DonateAll } from './DonateAll';

describe('DonateAll component', () => {
  test('renders DonateAll component', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const component = render(
      <Router>
        <DonateAll />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

  test('should render the TitleMain component', () => {
    render(
      <Router>
        <DonateAll />
      </Router>
    );
    const titleMainComponent = screen.getByRole('heading', { name: /Маленькі кроки ведуть нас до великої ПЕРЕМОГИ!/i });
    expect(titleMainComponent).toBeInTheDocument();
  });

  test('should render the TitleSection', () => {
    render(
      <Router>
        <DonateAll />
      </Router>
    );
    const titleSection = screen.getByRole('heading', { name: /ДОПОМОГТИ/i });
    expect(titleSection).toBeInTheDocument();
  });

  test('renders DonateAll component with 7 images (pay elements)', () => {
    render(
      <Router>
        <DonateAll />
      </Router>
    );
    const images = screen.getAllByRole('img');
    expect(images).toHaveLength(7);
  });
});
