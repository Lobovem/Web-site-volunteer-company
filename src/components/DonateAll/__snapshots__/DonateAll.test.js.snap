// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonateAll component renders DonateAll component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="donateAll"
      >
        <div
          class="donateAll__wrap"
        >
          <h2
            class="titleMain"
          >
            Маленькі кроки ведуть нас до великої ПЕРЕМОГИ!
          </h2>
          <h2
            class="donateAll__title"
          >
            Ми не збираємо кошти і не просимо донати в будь-яких соцмережах. Все робимо своїми силами. Але, якщо хочете допомогти, ми не можемо відмовити вам в такому праві.
          </h2>
          <div
            class="donateAll__wrapPay"
          >
            <div
              class="donateAll__wrapIcon"
            >
              <img
                alt="icon-privat-bank.svg"
                class="donateAll__img"
                src="./img/icon-privat-bank.svg"
              />
              <p
                class="donateAll__text"
              >
                Рахунок (Гривня)
              </p>
              <p
                class="donateAll__desc"
              >
                БО МБФ УКРАЇНСЬКИЙ ГУМАНІТАРНИЙ БАТАЛЬЙОН
              </p>
              <p
                class="donateAll__text"
              >
                ІПН/ЄДРПОУ
              </p>
              <p
                class="donateAll__desc"
              >
                44746691
              </p>
              <p
                class="donateAll__text"
              >
                IBAN
              </p>
              <p
                class="donateAll__desc"
              >
                UA533052990000026004031220892
              </p>
            </div>
            <div
              class="donateAll__wrapIcon"
            >
              <img
                alt="icon-mono.png"
                class="donateAll__img"
                src="./img/icon-mono.png"
              />
              <p
                class="donateAll__text"
              >
                Карта (Гривна)
              </p>
              <p
                class="donateAll__desc"
              >
                3758640000000000
              </p>
            </div>
            <div
              class="donateAll__wrapIcon"
            >
              <img
                alt="icon-swift.svg"
                class="donateAll__img"
                src="./img/icon-swift.svg"
              />
              <p
                class="donateAll__text"
              >
                SWIFT (Прекази у гривнях)
              </p>
              <p
                class="donateAll__desc"
              >
                IBAN UA00 000000 00000 00000000000000
              </p>
            </div>
            <div
              class="donateAll__wrapIcon"
            >
              <img
                alt="icon-pay-pal.svg"
                class="donateAll__img"
                src="./img/icon-pay-pal.svg"
              />
              <p
                class="donateAll__text"
              >
                Перейти до PayPal
              </p>
              <p
                class="donateAll__desc"
              >
                namecompany@gmail.com
              </p>
            </div>
            <div
              class="donateAll__wrapIconAlt"
            >
              <img
                alt="icon-bitcoin.svg"
                class="donateAll__img"
                src="./img/icon-bitcoin.svg"
              />
              <div
                class="donateAll__wrapText"
              >
                <p
                  class="donateAll__text"
                >
                  BTC
                </p>
                <p
                  class="donateAll__desc"
                >
                  0x0000000000000000000000000000000000000000
                </p>
              </div>
              <img
                alt="icon-ethereum.svg"
                class="donateAll__img"
                src="./img/icon-ethereum.svg"
              />
              <div
                class="donateAll__wrapText"
              >
                <p
                  class="donateAll__text"
                >
                  ETH
                </p>
                <p
                  class="donateAll__desc"
                >
                  0x00000000000000000000000000000000
                </p>
              </div>
              <img
                alt="icon-tether.svg"
                class="donateAll__img"
                src="./img/icon-tether.svg"
              />
              <div
                class="donateAll__wrapText"
              >
                <p
                  class="donateAll__text"
                >
                  USDT
                </p>
                <p
                  class="donateAll__desc"
                >
                  TG00000000000000000000000000000000
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="donateAll"
    >
      <div
        class="donateAll__wrap"
      >
        <h2
          class="titleMain"
        >
          Маленькі кроки ведуть нас до великої ПЕРЕМОГИ!
        </h2>
        <h2
          class="donateAll__title"
        >
          Ми не збираємо кошти і не просимо донати в будь-яких соцмережах. Все робимо своїми силами. Але, якщо хочете допомогти, ми не можемо відмовити вам в такому праві.
        </h2>
        <div
          class="donateAll__wrapPay"
        >
          <div
            class="donateAll__wrapIcon"
          >
            <img
              alt="icon-privat-bank.svg"
              class="donateAll__img"
              src="./img/icon-privat-bank.svg"
            />
            <p
              class="donateAll__text"
            >
              Рахунок (Гривня)
            </p>
            <p
              class="donateAll__desc"
            >
              БО МБФ УКРАЇНСЬКИЙ ГУМАНІТАРНИЙ БАТАЛЬЙОН
            </p>
            <p
              class="donateAll__text"
            >
              ІПН/ЄДРПОУ
            </p>
            <p
              class="donateAll__desc"
            >
              44746691
            </p>
            <p
              class="donateAll__text"
            >
              IBAN
            </p>
            <p
              class="donateAll__desc"
            >
              UA533052990000026004031220892
            </p>
          </div>
          <div
            class="donateAll__wrapIcon"
          >
            <img
              alt="icon-mono.png"
              class="donateAll__img"
              src="./img/icon-mono.png"
            />
            <p
              class="donateAll__text"
            >
              Карта (Гривна)
            </p>
            <p
              class="donateAll__desc"
            >
              3758640000000000
            </p>
          </div>
          <div
            class="donateAll__wrapIcon"
          >
            <img
              alt="icon-swift.svg"
              class="donateAll__img"
              src="./img/icon-swift.svg"
            />
            <p
              class="donateAll__text"
            >
              SWIFT (Прекази у гривнях)
            </p>
            <p
              class="donateAll__desc"
            >
              IBAN UA00 000000 00000 00000000000000
            </p>
          </div>
          <div
            class="donateAll__wrapIcon"
          >
            <img
              alt="icon-pay-pal.svg"
              class="donateAll__img"
              src="./img/icon-pay-pal.svg"
            />
            <p
              class="donateAll__text"
            >
              Перейти до PayPal
            </p>
            <p
              class="donateAll__desc"
            >
              namecompany@gmail.com
            </p>
          </div>
          <div
            class="donateAll__wrapIconAlt"
          >
            <img
              alt="icon-bitcoin.svg"
              class="donateAll__img"
              src="./img/icon-bitcoin.svg"
            />
            <div
              class="donateAll__wrapText"
            >
              <p
                class="donateAll__text"
              >
                BTC
              </p>
              <p
                class="donateAll__desc"
              >
                0x0000000000000000000000000000000000000000
              </p>
            </div>
            <img
              alt="icon-ethereum.svg"
              class="donateAll__img"
              src="./img/icon-ethereum.svg"
            />
            <div
              class="donateAll__wrapText"
            >
              <p
                class="donateAll__text"
              >
                ETH
              </p>
              <p
                class="donateAll__desc"
              >
                0x00000000000000000000000000000000
              </p>
            </div>
            <img
              alt="icon-tether.svg"
              class="donateAll__img"
              src="./img/icon-tether.svg"
            />
            <div
              class="donateAll__wrapText"
            >
              <p
                class="donateAll__text"
              >
                USDT
              </p>
              <p
                class="donateAll__desc"
              >
                TG00000000000000000000000000000000
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
